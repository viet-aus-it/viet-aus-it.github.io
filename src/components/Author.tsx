/* eslint-disable react/no-danger */
import { GatsbyImage } from 'gatsby-plugin-image';
import { rhythm } from '../utils/typography';

interface AuthorProps {
  author: ContentfulAuthor;
}

function Author({ author }: AuthorProps) {
  const avatar = author.image.gatsbyImageData;

  return (
    <div
      style={{
        display: `flex`,
        marginBottom: rhythm(2.5),
      }}
    >
      <GatsbyImage
        image={{
          ...avatar,
          layout: 'fixed',
          width: 150,
          height: 150,
        }}
        alt={author.name}
        style={{
          marginRight: rhythm(1 / 2),
          marginBottom: 0,
          minWidth: 50,
          borderRadius: `100%`,
        }}
      />
      <p>
        {`Written by `}
        <strong>{author.name}</strong>
        <span
          dangerouslySetInnerHTML={{
            __html: author.shortBio.childMarkdownRemark.html,
          }}
        />
      </p>
    </div>
  );
}

export default Author;
