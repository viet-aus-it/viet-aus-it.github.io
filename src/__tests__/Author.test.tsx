import { render } from '@testing-library/react';
import Author from '../components/Template/Author';

describe('Author view', () => {
  it('Should render "proper" author node', () => {
    const author = {
      name: 'Vietnamese Aussies in IT',
      title: 'Non-profit Organization',
      company: 'VAIT',
      shortBio: {
        shortBio:
          "We're a group of IT Geeks in Australia.\n\nCome find us on [GitHub](https://github.com/viet-aus-it/), [Discord](https://discord.gg/waYhnk4NKy), Facebook [group](https://www.facebook.com/group/vietausit) and [page](https://www.facebook.com/vietausit).",
        childMarkdownRemark: {
          html:
            '<p>Weâ€™re a group of IT Geeks in Australia.</p>\n<p>Come find us on <a href="https://github.com/viet-aus-it/">GitHub</a>, <a href="https://discord.gg/waYhnk4NKy">Discord</a>, Facebook <a href="https://www.facebook.com/group/vietausit">group</a> and <a href="https://www.facebook.com/vietausit">page</a>.</p>',
        },
      },
      github: 'viet-aus-it',
      image: {
        gatsbyImageData: {
          images: {
            sources: [
              {
                srcSet:
                  '//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=750&h=500&q=50&fm=webp 750w,\n//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=1500&h=1000&q=50&fm=webp 1500w,\n//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=3000&h=2000&q=50&fm=webp 3000w',
                sizes: '(min-width: 3000px) 3000px, 100vw',
                type: 'image/webp',
              },
            ],
            fallback: {
              src:
                '//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=3000&h=2000&fl=progressive&q=50&fm=jpg',
              srcSet:
                '//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=750&h=500&fl=progressive&q=50&fm=jpg 750w,\n//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=1500&h=1000&fl=progressive&q=50&fm=jpg 1500w,\n//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=3000&h=2000&fl=progressive&q=50&fm=jpg 3000w',
              sizes: '(min-width: 3000px) 3000px, 100vw',
            },
          },
          layout: 'constrained',
          width: 3000,
          height: 2000,
          backgroundColor: '#FFF',
        },
      },
    };

    const tree = render(<Author author={author} />);
    expect(tree).toMatchSnapshot();
  });
});
