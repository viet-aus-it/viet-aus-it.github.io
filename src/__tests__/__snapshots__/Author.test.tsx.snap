// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author view Should render "proper" author node 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; margin-bottom: 4.375rem;"
      >
        <div
          class="gatsby-image-wrapper"
          data-gatsby-image-wrapper=""
          style="position: relative; overflow: hidden; width: 150px; height: 150px; margin-right: 0.875rem; margin-bottom: 0px; min-width: 50px; border-radius: 100%;"
        >
          <div
            aria-hidden="true"
            data-placeholder-image=""
            style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
          />
          <picture>
            <source
              data-srcset="//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=750&h=500&q=50&fm=webp 750w,//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=1500&h=1000&q=50&fm=webp 1500w,//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=3000&h=2000&q=50&fm=webp 3000w"
              sizes="(min-width: 3000px) 3000px, 100vw"
              type="image/webp"
            />
            <img
              alt="Vietnamese Aussies in IT"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=3000&h=2000&fl=progressive&q=50&fm=jpg"
              data-srcset="//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=750&h=500&fl=progressive&q=50&fm=jpg 750w,//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=1500&h=1000&fl=progressive&q=50&fm=jpg 1500w,//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=3000&h=2000&fl=progressive&q=50&fm=jpg 3000w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 3000px) 3000px, 100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
          </picture>
          <noscript />
          <script
            type="module"
          >
            const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
          </script>
        </div>
        <p>
          Written by 
          <strong>
            Vietnamese Aussies in IT
          </strong>
          <span>
            <p>
              We’re a group of IT Geeks in Australia.
            </p>
            

            <p>
              Come find us on 
              <a
                href="https://github.com/viet-aus-it/"
              >
                GitHub
              </a>
              , 
              <a
                href="https://discord.gg/waYhnk4NKy"
              >
                Discord
              </a>
              , Facebook 
              <a
                href="https://www.facebook.com/group/vietausit"
              >
                group
              </a>
               and 
              <a
                href="https://www.facebook.com/vietausit"
              >
                page
              </a>
              .
            </p>
          </span>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; margin-bottom: 4.375rem;"
    >
      <div
        class="gatsby-image-wrapper"
        data-gatsby-image-wrapper=""
        style="position: relative; overflow: hidden; width: 150px; height: 150px; margin-right: 0.875rem; margin-bottom: 0px; min-width: 50px; border-radius: 100%;"
      >
        <div
          aria-hidden="true"
          data-placeholder-image=""
          style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        />
        <picture>
          <source
            data-srcset="//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=750&h=500&q=50&fm=webp 750w,//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=1500&h=1000&q=50&fm=webp 1500w,//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=3000&h=2000&q=50&fm=webp 3000w"
            sizes="(min-width: 3000px) 3000px, 100vw"
            type="image/webp"
          />
          <img
            alt="Vietnamese Aussies in IT"
            data-gatsby-image-ssr=""
            data-main-image=""
            data-src="//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=3000&h=2000&fl=progressive&q=50&fm=jpg"
            data-srcset="//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=750&h=500&fl=progressive&q=50&fm=jpg 750w,//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=1500&h=1000&fl=progressive&q=50&fm=jpg 1500w,//images.ctfassets.net/yy5lveaxjicl/7orLdboQQowIUs22KAW4U/8062cc92daeecfeb308f317f7f7d3dd9/matt-palmer-254999.jpg?w=3000&h=2000&fl=progressive&q=50&fm=jpg 3000w"
            decoding="async"
            loading="lazy"
            sizes="(min-width: 3000px) 3000px, 100vw"
            style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
          />
        </picture>
        <noscript />
        <script
          type="module"
        >
          const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
        </script>
      </div>
      <p>
        Written by 
        <strong>
          Vietnamese Aussies in IT
        </strong>
        <span>
          <p>
            We’re a group of IT Geeks in Australia.
          </p>
          

          <p>
            Come find us on 
            <a
              href="https://github.com/viet-aus-it/"
            >
              GitHub
            </a>
            , 
            <a
              href="https://discord.gg/waYhnk4NKy"
            >
              Discord
            </a>
            , Facebook 
            <a
              href="https://www.facebook.com/group/vietausit"
            >
              group
            </a>
             and 
            <a
              href="https://www.facebook.com/vietausit"
            >
              page
            </a>
            .
          </p>
        </span>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
